libraries/AP_IOMCU/iofirmware/iofirmware.cpp

```diff
@@ -171,6 +171,15 @@ void AP_IOMCU_FW::init()
         has_heater = true;
     }
 
+    //Set Heater PWM Polarity, 0 for Active Low and 1 for Active High
+    heater_pwm_polarity = !palReadLine(HAL_GPIO_PIN_HEATER);
```

Comment:
can you explain how this works, especially given the change to use pullup on the pin?